src/controller_examples/fluent_controller/fluentbit/exec/resource/service.rs

```diff
@@ -119,8 +119,16 @@ pub fn make_service(fb: &FluentBit) -> (service: Service)
         let mut metadata = ObjectMeta::default();
         metadata.set_name(make_service_name(fb));
         metadata.set_owner_references(make_owner_references(fb));
-        metadata.set_labels(make_labels(fb));
-        metadata.set_annotations(fb.spec().annotations());
+        metadata.set_labels({
+            if fb.spec().service_labels().is_some() {
+                fb.spec().service_labels().unwrap()
+            } else { make_labels(fb) }
+        });
+        if fb.spec().service_annotations().is_some() {
```

Comment:
For the service annotations, I think we can just set it to `fb.spec().service_annotations()`, similar to what you did for service account.